vm/src/stdlib/cmath.rs

```diff
@@ -0,0 +1,163 @@
+    use crate::{PyResult, VirtualMachine};
+    use num_complex::Complex64;
+
+    /// Return argument, also known as the phase angle, of a complex.
+    #[pyfunction]
+    fn phase(z: IntoPyComplex) -> f64 {
+        z.to_complex().arg()
+    }
+
+    /// Convert a complex from rectangular coordinates to polar coordinates.
+    ///
+    /// r is the distance from 0 and phi the phase angle.
+    #[pyfunction]
+    fn polar(x: IntoPyComplex) -> (f64, f64) {
+        x.to_complex().to_polar()
+    }
+
+    /// Convert from polar coordinates to rectangular coordinates.
+    #[pyfunction]
+    fn rect(r: IntoPyFloat, phi: IntoPyFloat) -> Complex64 {
+        Complex64::from_polar(r.to_f64(), phi.to_f64())
+    }
+
+    /// Checks if the real or imaginary part of z is infinite.
+    #[pyfunction]
+    fn isinf(z: IntoPyComplex) -> bool {
+        let Complex64 { re, im } = z.to_complex();
+        re.is_infinite() || im.is_infinite()
+    }
+
+    /// Return True if both the real and imaginary parts of z are finite, else False.
+    #[pyfunction]
+    fn isfinite(z: IntoPyComplex) -> bool {
+        z.to_complex().is_finite()
+    }
+
+    /// Checks if the real or imaginary part of z not a number (NaN)..
+    #[pyfunction]
+    fn isnan(z: IntoPyComplex) -> bool {
+        z.to_complex().is_nan()
+    }
+
+    /// Return the exponential value e**z.
+    #[pyfunction]
+    fn exp(z: IntoPyComplex, vm: &VirtualMachine) -> PyResult<Complex64> {
+        let z = z.to_complex();
+        result_or_overflow(z, z.exp(), vm)
+    }
+
+    #[derive(FromArgs)]
+    struct IsCloseArgs {
+        #[pyarg(positional)]
+        a: IntoPyComplex,
+        #[pyarg(positional)]
+        b: IntoPyComplex,
+        #[pyarg(named, optional)]
+        rel_tol: OptionalArg<IntoPyFloat>,
+        #[pyarg(named, optional)]
+        abs_tol: OptionalArg<IntoPyFloat>,
+    }
+
+    /// Determine whether two complex numbers are close in value.
+    ///
+    ///   rel_tol
+    ///     maximum difference for being considered "close", relative to the
+    ///     magnitude of the input values
+    ///   abs_tol
+    ///     maximum difference for being considered "close", regardless of the
+    ///     magnitude of the input values
+    ///
+    /// Return True if a is close in value to b, and False otherwise.
+    ///
+    /// For the values to be considered close, the difference between them must be
+    /// smaller than at least one of the tolerances.
+    ///
+    /// -inf, inf and NaN behave similarly to the IEEE 754 Standard. That is, NaN is
+    /// not close to anything, even itself. inf and -inf are only close to themselves.
+    #[pyfunction]
+    fn isclose(args: IsCloseArgs, vm: &VirtualMachine) -> PyResult<bool> {
+        let a = args.a.to_complex();
+        let b = args.b.to_complex();
+        let rel_tol = match args.rel_tol {
+            OptionalArg::Missing => 1e-09,
+            OptionalArg::Present(ref value) => value.to_f64(),
+        };
```

Comment:
`OptionalArg` supports `Option` methods.
```suggestion
        let rel_tol = args.rel_tol.map_or(1e-09, |value| value.to_f64());
```